File: app/src/main/java/com/sioptik/main/apriltag/AprilTagNative.java
package com.sioptik.main.apriltag;

import android.graphics.Bitmap;

import java.util.ArrayList;

/**
 * Interface to native C AprilTag library.
 */

public class AprilTagNative {
    static {
        System.loadLibrary("apriltag");
        native_init();
    }

    public static native void native_init();

    public static native void yuv_to_rgb(byte[] src, int width, int height, Bitmap dst);

    public static native void apriltag_init(String tagFamily, int errorBits, double decimateFactor,
                                            double blurSigma, int nthreads);

    public static native ArrayList<AprilTagDetection> apriltag_detect_yuv(byte[] src, int width, int height);
}

File: app/src/main/java/com/example/sioptik/ApriltagNative.kt
package com.example.sioptik

import android.graphics.Bitmap


class ApriltagNative {
    init {
        System.loadLibrary("apriltag")
        native_init()
    }

    external fun native_init()
    external fun yuv_to_rgb(src: ByteArray?, width: Int, height: Int, dst: Bitmap?)
    external fun apriltag_init(
        tagFamily: String?, errorBits: Int, decimateFactor: Double,
        blurSigma: Double, nthreads: Int
    )

    external fun apriltag_detect_yuv(
        src: ByteArray?,
        width: Int,
        height: Int
    ): ArrayList<ApriltagDetection?>?
}
